ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -1835,26 +1883,37 @@ def flatten_fw_and_to_np(*, ret, fw):
     return ret_np_flat
 
 
-def flatten(*, ret):
+def flatten(*, ret, frontend=None):
```

Comment:
I'd suggest creating a separate function for the frontend flattening, the test functions gets called a lot during the test execution, less branching is better not to add an overhead, also in this way it's clear that the frontends and backends have separate test logic

Reply:
yes sure. Wrote flatten_frontend instead.